### PR TITLE
Handle missing GitHub forks, add gem flexibility

### PR DESCRIPTION
**Pull request by akofink**
- Specify your own repos to clone.
- Clone repos without forks.
- Specify local gems to enable
- Run hammer from hammer-cli-foreman instead of hammer-cli-katello

Thanks to @johnpmitsch for suggesting improvements.

### CODE REVIEW

File: roles/hammer_devel/defaults/main.yml [hunk 1]
```diff
@@ -14,3 +14,9 @@ hammer_devel_repositories:
   - theforeman/hammer-cli-foreman-tasks
   - katello/hammer-cli-csv
   - katello/hammer-cli-katello
+hammer_devel_local_gems: |-
+  gem 'hammer_cli'
+    gem 'hammer_cli_katello'
```
Comment (ekohl) [inline roles/hammer_devel/defaults/main.yml]: Intending looks a bit off

File: roles/hammer_devel/tasks/hammer_install.yml [hunk 2]
```diff
@@ -7,33 +7,43 @@
...
+  changed_when: False
   with_items: "{{ hammer_devel_repositories }}"
-  register: fork_remotes_exist
+  register: local_fork_remotes_exist
 
-- name: 'Add fork remotes to cloned repositories'
-  command: "git remote add {{ hammer_devel_github_fork_remote_name }} git@github.com:{{ hammer_devel_github_username }}/{{ item.item.split('/')[1] }}.git"
-  when: item.rc != 0
+- name: 'Check the GitHub fork repos exist for missing local fork remotes'
+  shell: "git ls-remote --exit-code -h https://:@github.com/{{ hammer_devel_github_username }}/{{ item.item.split('/')[1] }}"
```
Comment (ekohl) [inline roles/hammer_devel/tasks/hammer_install.yml]: Is the `:@` needed in the URL? Forces that to not ask for credentials on missing/private repos?

File: vagrant/boxes.d/99-local.yaml.example [hunk 3]
```diff
@@ -25,9 +25,18 @@ centos7-hammer-devel:
   ansible:
     playbook: 'playbooks/hammer_devel.yml'
     group: 'hammer-devel'
-    variables:
-      hammer_devel_github_username: '<REPLACE ME>'
-      hammer_devel_github_fork_remote_name: 'origin'
+    variables: # defaults in roles/hammer_devel/deafults/main.yml
```
Comment (ekohl) [inline vagrant/boxes.d/99-local.yaml.example]: deafults -> defaults

File: vagrant/boxes.d/99-local.yaml.example [hunk 4]
```diff
@@ -25,9 +25,18 @@ centos7-hammer-devel:
...
+      hammer_devel_github_username: <REPLACE ME>
+      hammer_devel_github_fork_remote_name: origin
+      hammer_devel_server: centos7-devel.custom-domain
+      hammer_devel_repositories:
+        - theforeman/hammer-cli
+        - theforeman/hammer-cli-foreman
+        - katello/hammer-cli-katello
+      hammer_devel_local_gems: |-
+        gem 'hammer_cli'
+          gem 'hammer_cli_foreman'
```
Comment (ekohl) [inline vagrant/boxes.d/99-local.yaml.example]: odd indentation

File: roles/hammer_devel/tasks/hammer_install.yml (hunk 2 above)
Comment (akofink) [reply roles/hammer_devel/tasks/hammer_install.yml]: Yep! It's interactive without the :@

File: roles/hammer_devel/defaults/main.yml (hunk 1 above)
Comment (akofink) [reply roles/hammer_devel/defaults/main.yml]: The indentation is correct to format the file correctly. Try it! :)

**Comment by akofink** [comment]
Fixed the typo

File: roles/hammer_devel/defaults/main.yml (hunk 1 above)
Comment (akofink) [reply roles/hammer_devel/defaults/main.yml]: I'll point you to this very confusing documentation http://yaml.org/spec/1.2/spec.html#id2777534

**Pull request merged**